frontend/src/plugins/impl/DataTablePlugin.tsx

```diff
@@ -122,20 +133,43 @@ interface DataTableProps extends Data<unknown>, Functions {
   className?: string;
   value: S;
   setValue: (value: S) => void;
+  sorting?: SortingState;
+  setSorting?: OnChangeFn<SortingState>;
 }
 
 export const LoadingDataTableComponent = memo(
   (props: DataTableProps & { data: string }) => {
+    const [tableData, setTableData] = useState(props.data);
+
+    useEffect(() => {
+      const getSortedData = async () => {
+        if (!props.sorting || props.sorting.length === 0) {
+          setTableData(props.data);
+          return;
+        }
+
+        const sortedData = await props.sort_values({
+          by: props.sorting[0].id,
+          descending: props.sorting[0].desc,
+        });
+
+        setTableData(sortedData);
+      };
+
+      getSortedData();
+    }, [props.sorting, props.sort_values]);
+
     const { data, loading, error } = useAsyncData<unknown[]>(() => {
-      if (!props.data || props.totalRows === 0) {
+      if (!tableData || props.totalRows === 0) {
         return Promise.resolve([]);
       }
       return vegaLoadData(
-        props.data,
+        tableData,
         { type: "csv", parse: getVegaFieldTypes(props.fieldTypes) },
-        { handleBigInt: true },
+        { handleBigInt: true }
       );
-    }, [props.data, props.fieldTypes, props.totalRows]);
+    }, [tableData, props.fieldTypes, props.totalRows]);
```

Comment:
Does this work? (combining the two)

```ts
    const { data, loading, error } = useAsyncData<unknown[]>(() => {
      const getSortedData = async () => {
        if (!props.sorting || props.sorting.length === 0) {
          return props.data;
        }

        const sortedData = await props.sort_values({
          by: props.sorting[0].id,
          descending: props.sorting[0].desc,
        });

        return sortedData;
      };

      getSortedData();

      if (props.totalRows === 0) {
        return Promise.resolve([]);
      }

      return vegaLoadData(
        await getSortedData(),
        { type: "csv", parse: getVegaFieldTypes(props.fieldTypes) },
        { handleBigInt: true }
      );
    }, [props.sorting, props.sort_values, props.fieldTypes, props.totalRows]);

Reply:
Can't call `await` inside the useAsyncData hook:

```
'await' expressions are only allowed within async functions and at the top levels of modules.ts(1308)
DataTablePlugin.tsx(145, 62): Did you mean to mark this function as 'async'?
```

Reply:
Sorry it should be `const { data, loading, error } = useAsyncData<unknown[]>(async () => {`